### PR TITLE
chore: make FirestoreException factory methods public to allow construction for testing

### PR DESCRIPTION
**Pull request by BenWhitehead**
* Expose each factory method at @BetaApi level ensuring no guarantee about api surface
* Expose getStatus() at @BetaApi up from package private @InternalApi

### CODE REVIEW

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java [hunk 1]
```diff
@@ -55,7 +55,8 @@ private FirestoreException(IOException exception, boolean retryable) {
    *
    * @return The FirestoreException
    */
-  static FirestoreException invalidState(String message, Object... params) {
+  @BetaApi
+  public static FirestoreException invalidState(String message, Object... params) {
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java:59]: Maybe s/invalidState/forInvalidArgument ?

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java [hunk 2]
```diff
@@ -65,7 +66,8 @@ static FirestoreException invalidState(String message, Object... params) {
    *
    * @return The FirestoreException
    */
-  static FirestoreException serverRejected(Status status, String message, Object... params) {
+  @BetaApi
+  public static FirestoreException serverRejected(Status status, String message, Object... params) {
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java:70]: Maybe forServerRejection?

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java [hunk 3]
```diff
@@ -85,7 +88,8 @@ static FirestoreException serverRejected(
    *
    * @return The FirestoreException
    */
-  static FirestoreException networkException(IOException exception, boolean retryable) {
+  @BetaApi
+  public static FirestoreException networkException(IOException exception, boolean retryable) {
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java:92]: Maybe forIoException?

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java [hunk 4]
```diff
@@ -94,7 +98,8 @@ static FirestoreException networkException(IOException exception, boolean retrya
    *
    * @return The FirestoreException
    */
-  static FirestoreException apiException(ApiException exception) {
+  @BetaApi
+  public static FirestoreException apiException(ApiException exception) {
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java:102]: Maybe forApiException?

**Review by schmidt-sebastian — APPROVED** [review]

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java (hunk 1 above)
Comment (BenWhitehead) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java]: Done

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java (hunk 2 above)
Comment (BenWhitehead) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java]: Done

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java (hunk 3 above)
Comment (BenWhitehead) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java]: Done

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java (hunk 4 above)
Comment (BenWhitehead) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreException.java]: Done

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**